core/src/main/java/org/owasp/dependencycheck/Engine.java

```diff
@@ -250,13 +251,31 @@ public List<Analyzer> getAnalyzers(AnalysisPhase phase) {
     }
 
     /**
-     * Adds a dependency.
+     * Adds a dependency. In some cases, when adding a virtual dependency, the
+     * method will identify if the virtual dependency was previously added and
+     * if so the existing dependency will be returned.
      *
      * @param dependency the dependency to add
-     */
-    public synchronized void addDependency(Dependency dependency) {
+     * @return the dependency added; in the case of virtual dependencies this
+     * may not be the same as the dependency passed in as an argument
+     */
+    public synchronized Dependency addDependency(Dependency dependency) {
+        if (dependency.isVirtual()) {
+            for (Dependency existing : dependencies) {
+                if (existing.isVirtual()
+                        && existing.getSha256sum() != null
+                        && existing.getSha256sum().equals(dependency.getSha256sum())
+                        && existing.getDisplayFileName() != null
+                        && existing.getDisplayFileName().equals(dependency.getDisplayFileName())
+                        && identifiersMatch(existing.getSoftwareIdentifiers(), dependency.getSoftwareIdentifiers())) {
+                    DependencyBundlingAnalyzer.mergeDependencies(existing, dependency, null);
+                    return existing;
+                }
+            }
+        }
         dependencies.add(dependency);
         dependenciesExternalView = null;
+        return dependency;
```

Comment:
```suggestion
    public synchronized void addDependency(Dependency dependency) {
        if (dependency.isVirtual()) {
            for (Dependency existing : dependencies) {
                if (existing.isVirtual()
                        && existing.getSha256sum() != null
                        && existing.getSha256sum().equals(dependency.getSha256sum())
                        && existing.getDisplayFileName() != null
                        && existing.getDisplayFileName().equals(dependency.getDisplayFileName())
                        && identifiersMatch(existing.getSoftwareIdentifiers(), dependency.getSoftwareIdentifiers())) {
                    DependencyBundlingAnalyzer.mergeDependencies(existing, dependency, null);
                    return;
                }
            }
        }
        dependencies.add(dependency);
        dependenciesExternalView = null;
```

As we never use the result there is no use in returning the dependency. We can just early-return from the method.